cloudsim-plus/src/main/java/org/cloudsimplus/faultinjection/VmCloner.java

```diff
@@ -0,0 +1,98 @@
+ * the creating of another Vm from an snapshot of the failed one.
+ * It also enables re-creating Cloudlets which were running
+ * inside the failed VM.
+ *
+ * @author raysaoliveira
+ * @since CloudSim Plus 1.2.2
+ */
+public interface VmCloner {
+    VmCloner NULL = new VmCloner() {
+        @Override public int getClonedVmsNumber() { return 0;}
+        @Override public Map.Entry<Vm, List<Cloudlet>> clone(Vm sourceVm) { return new HashMap.SimpleEntry<>(Vm.NULL, Collections.EMPTY_LIST); }
+        @Override public VmCloner setVmClonerFunction(UnaryOperator<Vm> vmClonerFunction) { return this; }
+        @Override public VmCloner setCloudletsClonerFunction(Function<Vm, List<Cloudlet>> cloudletsClonerFunction) { return this; }
+        @Override public int getMaxClonesNumber() { return 0; }
+        @Override public boolean isMaxClonesNumberReached() { return false; }
+        @Override public VmCloner setMaxClonesNumber(int maxClonesNumber) { return this; }
+    };
+
+    /**
+     * Gets the number of VMs cloned so far.
+     * @return
+     */
+    int getClonedVmsNumber();
+
+    /**
+     * Clones a given {@link Vm} using the Vm Cloner Function
+     * and their Cloudlets using the Clodlets Cloner Function,
+     * binding the cloned Cloudlets to the cloned Vm.
+     *
+     * @param sourceVm the Vm to be cloned
+     * @return a {@link Map.Entry} where the key is the cloned Vm
+     * and the value is the List of cloned Cloudltes.
+     * @see #setVmClonerFunction(UnaryOperator)
+     * @see #setCloudletsClonerFunction(Function)
+     */
+    Map.Entry<Vm, List<Cloudlet>> clone(Vm sourceVm);
+
+    /**
+     * Sets the {@link UnaryOperator} to be used to clone {@link Vm}s.
+     * It is a Function which, when called, creates a clone of a specific Vm.
+     * @param vmClonerFunction the {@link Vm} cloner Function to set
+     * @return
+     */
+    VmCloner setVmClonerFunction(UnaryOperator<Vm> vmClonerFunction);
+
+    /**
+     * Gets the {@link Function} to be used to clone Vm's {@link Cloudlet}s.
+     * It is a Function which, when called, creates a clone of cloudlets
+     * which were running inside a specific Vm.
+     *
+     * <p>Such a Function is used to recreate those Cloudlets
+     * inside a clone of the failed VM. In this case, all the Cloudlets are
+     * recreated from scratch into the cloned VM.
+     * This way, when they are submitted to a broker,
+     * they re-start execution from the beginning.
+     * </p>
+     *
+     * @param cloudletsClonerFunction the {@link Cloudlet}s cloner Function to set
+     * @return
+     */
+    VmCloner setCloudletsClonerFunction(Function<Vm, List<Cloudlet>> cloudletsClonerFunction);
+
+    /**
+     * Gets the maximum number of Vm clones to create.
+     * For instance, if this value is equal to 2,
+     * it means if all VMs from a given broker are destroyed multiple times,
+     * a clone will be created only 2 times. For the 3rd time all VMs are destroyed again,
```

Comment:
Change "For the 3rd time all VMs are destroyed again, no clone will be created." to "If all VMs are destroyed again for the 3rd time, no clone will be created."